### PR TITLE
Make istioctl install output more user friendly

### PR DESCRIPTION
**Pull request by ostromart**
Changes progress output to be more user facing, from this:
```
$ ic install --charts manifests --set profile=demo
- Processing resources for components Base.    
✔ Component Base installed                                                                                                                                                                     
✔ Component Pilot installed                                                                                                                                                                    
✔ Component EgressGateways installed                                                                                                                                                           
✔ Component IngressGateways installed                                                                                                                                                          
✔ Component AddonComponents installed                                                                                                                                                          
- Pruning removed resources                                                                                                                                                                    Pruned object PodDisruptionBudget:istio-system:istio-egressgateway.
Pruned object Deployment:istio-system:istio-egressgateway.
✔ Installation complete                                     
```
to this:
 ```
$ ic install --charts manifests --set profile=default
- Processing resources for Istio core.  
✔ Istio core installed                                                                                                                                                                         
✔ Istiod installed                                                                                                                                                                             
✔ Addons installed                                                                                                                                                                             
✔ Ingress gateways installed                                                                                                                                                                   
- Pruning removed resources                                                                                                                                                                    Removed PodDisruptionBudget:istio-system:istio-egressgateway.
Removed Deployment:istio-system:istio-egressgateway.
✔ Installation complete                      
```

### CODE REVIEW

**Review by elfinhe — APPROVED** [review]

**Review by elfinhe — APPROVED** [review]
LGTM if np lint is fixed

File: operator/pkg/helmreconciler/apply.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ import (
 	"istio.io/istio/operator/pkg/name"
 	"istio.io/istio/operator/pkg/object"
 	"istio.io/istio/operator/pkg/util"
+	"istio.io/istio/operator/pkg/util/progress"
```
Comment (howardjohn) [inline operator/pkg/helmreconciler/apply.go:34]: thanks, this is a nice change

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
```
- Pruning removed resources
Removed Deployment:istio-system:istio-egressgateway.
```

should be 

```
- Pruning removed resources
- Removed Deployment:istio-system:istio-egressgateway.
```

or something probably or something else?

**Comment by ostromart** [comment]
Changed to:
```
- Pruning removed resources
  Removed Deployment:istio-system:istio-egressgateway.
```

**Comment by jacob-delgado** [comment]
cc @ostromart @brian-avery @irisdingbj 

The cherry-pick failed and does not have a manual cherry picked PR for 1.6. I see that @ostromart has a commit that references this, but there is no PR yet. This is being used for documentation purposes.

**Pull request merged**